utils/pubsub_test.go

```diff
@@ -0,0 +1,144 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+)
+
+func TestPubSub_single_slow_consumer(t *testing.T) {
+	const topic = "topic"
+	bus := EventBus{}
+	slow := testConsumer{}
+	defer func() { slow.done <- true }() // stop consumer
+
+	slow.subscribe(topic, &bus)
+	bus.Publish(topic, 1)
+	<-slow.started // wait consumer to receive the 1st message
+	bus.Publish(topic, 2)
+	bus.Publish(topic, 3)
+	bus.Publish(topic, 4)
+	bus.Publish(topic, 5)
+
+	v := <-slow.out
+	if v != 1 {
+		t.Errorf("Expected slow subscriber to have only consumed first value 1, instead got %d.", v)
+	}
+	v = <-slow.out
+	if v != 5 {
+		t.Errorf("Expected slow subscriber to consume latest value 5, instead got %d.", v)
+	}
+}
+
+func TestPubSub_two_consumers_slow_and_fast(t *testing.T) {
+	const topic = "topic"
+	bus := EventBus{}
+
+	slow := testConsumer{}
+	slow.subscribe(topic, &bus)
+	defer func() { slow.done <- true }() // stop consumer
```

Comment:
There might be a deadlock as this line is preventing the test to complete when running it with the `-race` flag.

```
go test -race ./utils -run TestPubSub_two_consumers_slow_and_fast -v -count 1
```